tests/combine_samples.C

```diff
@@ -0,0 +1,200 @@
+#include <stdio.h>
+
+int main(int argc, char* argv[])
+{
+    // Initialize MPI
+    MPI_Init(&argc, &argv);
+    int size;
+    MPI_Comm_size(MPI_COMM_WORLD, &size);
+    int rank;
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+	
+    std::vector<std::string> sample_names;
+    int snaps = 0; 
+    int dim   = 0;
+    bool subtract_mean = false;
+    bool subtract_offset = false;
+    std::string kind = "snapshot";
+    std::string offset_file;
+
+    bool offset_arg = false;	
+
+    if (argc >= 2) {
+        for (int i = 1; i < argc; i++) {
+	    if (!strcmp(argv[i], "basis") || !strcmp(argv[i], "-b")) { 
+            if (rank==0) std::cout << "Argument " << i << " identified as basis or -b" << std::endl;
+                kind = "basis"; 
+            } 
+            else if (offset_arg) {
+                offset_file = argv[i];
+                offset_arg = false;
+            }
+            else if (!strcmp(argv[i], "offset") || !strcmp(argv[i], "-o")) {
+                if (rank==0) std::cout << "Will subtract offset" << std::endl;
+                subtract_offset = true;
+                offset_arg = true;
+            }
+            else if (!strcmp(argv[i], "mean") || !strcmp(argv[i], "-m")) {
+                if (rank==0) std::cout << "Will subtract mean" << std::endl;
+	    	subtract_mean = true;
+	    }
+	    else {
+                sample_names.push_back(argv[i]); 
+	    }
+	}
+    }
+    else {
+        if (rank==0) std::cout << "No arguments passed." << std::endl;
+        return 1;
+    }
+ 
+    /*-- Read dimension and count number of snapshots/bases --*/
+    if (rank==0) std::cout << "Opening files to read dimension and count number of snapshots/bases" << std::endl;
+    for (const auto& sample_name: sample_names) {
+        CAROM::BasisReader reader(sample_name);
+		
+        if (kind == "snapshot") {
+	    CAROM::Matrix *snapshots = (CAROM::Matrix*) reader.getSnapshotMatrix(0);
+	    dim    = snapshots->numRows();
+ 	    snaps += snapshots->numColumns();
+        }
+	else {
+            CAROM::Matrix *basis = (CAROM::Matrix*) reader.getSpatialBasis(0);
+	    dim    = basis->numRows();
+	    snaps += basis->numColumns();
+	}
+    }
+	
+    CAROM_VERIFY((snaps > 0) && (dim > 0));
+	
+    /*-- Load data from input files --*/
+    std::string generator_filename = "total";
+    std::unique_ptr<CAROM::BasisGenerator> static_basis_generator;
+    static_basis_generator.reset(new CAROM::BasisGenerator(
+                                     CAROM::Options(dim, snaps).setMaxBasisDimension(snaps), false,
+                                     generator_filename));
+
+    if (rank==0) std::cout << "Loading data from " << kind << std::endl;
+    for(const auto& sample_name: sample_names) {
+        static_basis_generator->loadSamples(sample_name, kind);
+    }
+
+    if (rank==0) std::cout << "Saving data uploaded as a snapshot" << std::endl;
+    static_basis_generator->writeSnapshot();
+	
+    if (!subtract_mean && !subtract_offset) {
```

Comment:
`CAROM_VERIFY(!(subtract_mean && subtract_offset));`

Reply:
It could be possible that the user doesn't want to subtract the mean or offset, so I've included that possibility here. Unless did you mean that I should include that verification inside this if statement?

Reply:
I meant the user should not specify both mean and offset.

Reply:
oh of course. Updated